### PR TITLE
Create 'Credentials' section of Controller UG

### PR DESCRIPTION
**Pull request by ianf77**
Adding files

Create Section 11 of the Controller user Guide

https://issues.redhat.com/browse/AAP-15822

### CODE REVIEW

File: downstream/modules/platform/ref-controller-credential-machine.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+This password will be stored encrypted in the database, if entered.
+Alternatively, you can configure {ControllerName} to ask the user for the password at launch time by selecting *Prompt on launch*. 
+In these cases, a dialog opens when the job is launched, promoting the user to enter the password and password confirmation.
+* *SSH Private Key*: Copy or drag-and-drop the SSH private key for the machine credential.
+* *Private Key Passphrase*: If the SSH Private Key used is protected by a password, you can configure a Key Password for the private key. 
+This password is stored encrypted in the database, if entered.
+Alternatively, you can configure {ControllerName} to ask the user for the password at launch time by selecting *Prompt on launch*. 
+In these cases, a dialog opens when the job is launched, prompting the user to enter the password and password confirmation.
+* *Privilege Escalation Method*: Specifies the type of escalation privilege to assign to specific users. 
+This is equivalent to specifying the `--become-method=BECOME_METHOD` parameter, where `BECOME_METHOD` is any of the following methods, or a custom method you've written.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-machine.adoc:29]:
```suggestion
This is equivalent to specifying the `--become-method=BECOME_METHOD` parameter, where `BECOME_METHOD` is any of the following methods, or a custom method you have written. 
```

File: downstream/modules/platform/ref-controller-credential-machine.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+//image:credentials-create-machine-credential-priv-escalation.png[image]
+
+** *empty selection*: If a task/play has `become` set to `yes` and is used with an empty selection, then it will default to `sudo`
+** *sudo*: Performs single commands with super user (root user) privileges
+** *su*: Switches to the super user (root user) account (or to other user accounts)
+** *pbrun*: Requests that an application or command be run in a controlled account and provides for advanced root privilege delegation and keylogging
+** *pfexec*: Executes commands with predefined process attributes, such as specific user or group IDs
+** *dzdo*: An enhanced version of sudo that uses RBAC information in an Centrify's Active Directory service. 
+For more information, see Centrify's link:http://community.centrify.com/t5/Centrify-Server-Suite/FAQ-What-is-DirectAuthorize-dzdo-dzwin/td-p/21193[site on DZDO]
+** *pmrun*: Requests that an application is run in a controlled account, see link:http://documents.software.dell.com/privilege-manager-for-unix/6.0/administrators-guide/privilege-manager-programs/pmrun[Privilege Manager for Unix 6.0]
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-machine.adoc:41]:
```suggestion
** *pmrun*: Requests that an application is run in a controlled account. See link:http://documents.software.dell.com/privilege-manager-for-unix/6.0/administrators-guide/privilege-manager-programs/pmrun[Privilege Manager for Unix 6.0]
```

**Review by ariordan-redhat — COMMENTED** [review]
First half of review

File: downstream/modules/platform/proc-controller-credential-create-openshift-account.adoc [hunk 3]
```diff
@@ -0,0 +1,38 @@
+[id="proc-controller-credential-create-openshift-account"]
+
+= Creating a service account in an Openshift cluster
+
+This section describes creating a Service Account in an Openshift cluster (or K8s) to be used to run jobs in a container group through {ControllerName}. 
+After the Service Account is created, its credentials are provided to {ControllerName} in the form of an Openshift or Kubernetes API bearer token credential. 
+
+Use the following procedure to creates a service account and collect the information required to configure {ControllerName}.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/proc-controller-credential-create-openshift-account.adoc:8]:
```suggestion
Use the following procedure to create a service account and collect the information required to configure {ControllerName}.
```

File: downstream/modules/platform/ref-controller-credential-GCE.adoc [hunk 4]
```diff
@@ -0,0 +1,28 @@
+[id="ref-controller-credential-GCE"]
+
+= Google Compute Engine
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-GCE.adoc:3]: There's another module with this title
10.4.9 and 11.2 in my local build

File: downstream/modules/platform/ref-controller-credential-GCE.adoc [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+These are fields prompted in the user interface:
+
+[literal, options="nowrap" subs="+attributes"]
+----
+GCE_EMAIL
+GCE_PROJECT
+GCE_CREDENTIALS_FILE_PATH
+----
+
+These are fields prompted in the user interface.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-GCE.adoc:17]:
```suggestion

```
Duplication

File: downstream/modules/platform/ref-controller-credential-aap.adoc [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[id="ref-controller-credential-aap"]
+
+= {PlatformName}
+
+Select this credential to access another {ControllerName} instance.
+
+//image:credentials-create-at-credential.png[Credentials- create tower credential]
+
+{PlatformNameStart} credentials require the following inputs:
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-aap.adoc:9]:
```suggestion
{PlatformNameShort} credentials require the following inputs:
```

File: downstream/modules/platform/ref-controller-credential-azure-resource.adoc [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+Select this credential type to enable synchronization of cloud inventory with {Azure} Resource Manager.
+
+//image:credentials-create-azure-credential.png[Credentials- create Azure credential]
+
+{Azure} Resource Manager credentials require the following inputs:
+
+* *Subscription ID*: The Subscription UUID for the {Azure} account (required).
+* *Username*: The username to use to connect to the {Azure} account.
+* *Password*: The password to use to connect to the (Azure) account.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-azure-resource.adoc:13]:
```suggestion
* *Password*: The password to use to connect to the {Azure} account.
```

File: downstream/modules/platform/ref-controller-credential-openShift.adoc [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+Select this credential type to create instance groups that point to a Kubernetes or OpenShift container. 
+
+For more information, see link:https://docs.ansible.com/automation-controller/4.4/html/administration/containers_instance_groups.html#ag-ext-exe-env[Container and Instance Groups] in the Automation controller Administration Guide.
+
+//image:credentials-create-containers-credential.png[Credentials- create Containers credential]
+
+Container credentials require the following inputs:
+
+* *OpenShift or Kubernetes API Endpoint* (required): the endpoint used to connect to an OpenShift or Kubernetes container
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-openShift.adoc:13]:
```suggestion
* *OpenShift or Kubernetes API Endpoint* (required): the endpoint used to connect to an OpenShift or Kubernetes container.
```

File: downstream/modules/platform/ref-controller-credential-openShift.adoc [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+* *API Authentication Bearer Token* (required): The token used to authenticate the connection
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-openShift.adoc:14]:
```suggestion
* *API Authentication Bearer Token* (required): The token used to authenticate the connection.
```

File: downstream/modules/platform/ref-controller-credential-openShift.adoc [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+Environments that use internal or private _Certificate Authority_ (CA) must leave this option unchecked to disable verification.
+* *Certificate Authority Data*: include the `BEGIN CERTIFICATE` and `END CERTIFICATE` lines when pasting the certificate, if provided.
+
+A ContainerGroup is a type of InstanceGroup that has an associated Credential that enables connection to an OpenShift cluster. 
+To set up a container group, you must have the following:
+
+* A namespace you can launch into (every cluster has a “default” namespace, but you may want to use a specific namespace).
+* A service account that has the roles that enable it to launch and manage Pods in this namespace.
+* If you use {ExecEnvShort}s in a private registry, and have a Container Registry credential associated with them in {ControllerName}, the service account also requires the roles to get, create, and delete secrets in the namespace. 
+
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-openShift.adoc:25]:
```suggestion
+
```
The bullet points are off 
Suggesting a "+", but you might want different formatting

File: downstream/modules/platform/ref-controller-credential-openStack.adoc [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+* *Username*: The username to use to connect to OpenStack.
+* *Password (API Key)*: The password or API key to use to connect to OpenStack.
+* *Host (Authentication URL)*: The host to be used for authentication.
+* *Project (Tenant Name)*: The Tenant name or Tenant ID used for OpenStack. 
+This value is usually the same as the username.
+* Optional: *Project (Domain Name)*: Provide the project name associated with your domain.
+* Optional: *Domain name*: Provide the FQDN to be used to connect to OpenStack.
+
+If you are interested in using OpenStack Cloud Credentials, see xref:controller-cloud-credentials[Use Cloud Credentials with a cloud inventory] in this guide includes a sample playbook.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-openStack.adoc:19]:
```suggestion
If you are interested in using OpenStack Cloud Credentials, see xref:controller-cloud-credentials[Use Cloud Credentials with a cloud inventory] in this guide, which includes a sample playbook.
```

File: downstream/modules/platform/ref-controller-credential-vault.adoc [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+//image:credentials-create-vault-credential.png[Credentials- create Vault credential]
+
+Vault credentials require the *Vault Password* and an optional *Vault Identifier* if applying multi-Vault credentialing. 
+
+For more information on the Multi-Vault support, refer to the [Multi-Vault Credentials] section of the
+_Automation Controller Administration Guide_.
+
+You can configure {ControllerName} to ask the user for the password at launch time by selecting *Prompt on launch*. 
+
+In these cases, a dialog opens when the job is launched, promoting the user to enter the password and password.
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-vault.adoc:16]:
```suggestion
When *Prompt on launch* is selected, a dialog opens when the job is launched, promoting the user to enter the password and password.
```

File: downstream/modules/platform/ref-controller-credential-virtualization.adoc [hunk 13]
```diff
@@ -0,0 +1,27 @@
+[id="ref-controller-credential-virtualization"]
+
+= Red Hat Virtualization
```
Comment (ariordan-redhat) [inline downstream/modules/platform/ref-controller-credential-virtualization.adoc:3]: Should we be including this, given that RHV is being discontinued. (I think) ?

**Review by ariordan-redhat — COMMENTED** [review]
2nd half of review

File: downstream/modules/platform/ref-controller-credential-centrify-vault.adoc [hunk 14]
```diff
@@ -0,0 +1,6 @@
+[id="ref-controller-credential-centrify-vault"]
+
+= Centrify Vault Credential Provider Lookup
+
+This is considered part of the secret management capability.
+For more information, see [Centrify Vault Credential Provider Lookup].
```
[the remark this answers is not in the document]
Comment (ianf77) [reply downstream/modules/platform/ref-controller-credential-centrify-vault.adoc:6]: Yes. Two sections from here, I'll update the xrefs once it's done.

File: downstream/modules/platform/ref-controller-credential-virtualization.adoc (hunk 13 above)
Comment (ianf77) [reply downstream/modules/platform/ref-controller-credential-virtualization.adoc:3]: Something for the SMEs to decide.

**Pull request merged**